.github/workflows/test_migrations.yaml

```diff
@@ -0,0 +1,245 @@
+
+    steps:
+      - uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache
+            ~/.local
+            ~/.npm
+          key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+
+      - name: Checkout master
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: master
+          submodules: true
+
+      - name: Install system dependencies
+        run: |
+          curl -sL https://deb.nodesource.com/setup_14.x | sudo -E bash -
+
+          sudo apt install -y wget nodejs unzip firefox nginx
+
+          pip install --upgrade pip
+          pip install wheel numpy
+
+          export NPM_PACKAGES="${HOME}/.npm-packages"
+          export PATH=${NPM_PACKAGES}/bin:$PATH
+          export NODE_PATH="$NPM_PACKAGES/lib/node_modules:$NODE_PATH"
+
+          sudo npm -g install npm@6.14.11
+
+          which python; python --version
+          echo npm $(npm --version)
+          echo node $(node --version)
+          nginx -v
+          firefox --version
+
+      - name: Install SkyPortal dependencies
+        run: |
+          export PYTHONPATH=$PYTHONPATH:$(pwd)
+          make dependencies
+
+          pip install black
+
+      - name: Save current Alembic head on master
+        run: |
+          # Get the head line from Alembic history output
+          head=$( PYTHONPATH=. alembic history | grep '(head)' )
+
+          # Split by space; index 2 should be latest revision
+          tokens=($head)
+          current_head=${tokens[2]}
+          echo ${current_head}
+          echo "CURRENT_HEAD=${current_head}" >> $GITHUB_ENV
+
+      - name: Checkout commit before the very first migration
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: c48fa7f040aa83a494a365f80a8dffeaa0af9a1f
+          submodules: true
+
+      - name: Write SkyPortal configs
+        run: |
+          cat << EOF > config.yaml
+            database:
+              database: skyportal
+              host: localhost
+              port: 5432
+              user: skyportal
+              password: anything
+          EOF
+
+      - name: Initialize SkyPortal
+        run: |
+          # Usually, we create databases on the local machine, so
+          # `createdb` just works out of the box.  However, when doing
+          # this on a server, as in our case, we need to specify extra
+          # options.
+          #
+          # db_init should not complain if the databases exist already
+          #
+          echo "localhost:5432:*:skyportal:anything" > ~/.pgpass
+          chmod 600 ~/.pgpass
+
+          createdb -h localhost -U skyportal skyportal_test
+          psql -U skyportal -h localhost -c "GRANT ALL PRIVILEGES ON DATABASE skyportal_test TO skyportal;" skyportal_test
+
+          make db_init
+
+          pip list --format=columns
+          npm ls --depth 0
```

Comment:
Why is this needed?

Reply:
It's not really necessary - I mostly copied this section from existing workflows. I guess this was added for potential debugging purposes - say if some dependency upgrade ends up breaking the build for whatever reason.